pennylane/labs/resource_estimation/ops/op_math/controlled_ops.py

```diff
@@ -291,37 +611,94 @@ def controlled_resource_decomp(
 
     @classmethod
     def pow_resource_decomp(cls, z) -> Dict[re.CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources for an operator raised to a power.
+
+        Args:
+            z (int): the power that the operator is being raised to
+
+        Resources:
+            This operation is self-inverse, thus when raised to even integer powers acts like
+            the identity operator and raised to odd powers it produces itself.
+
+        Returns:
+            Dict[CompressedResourceOp, int]: The keys are the operators and the associated
+                values are the counts.
+        """
         return {} if z % 2 == 0 else {cls.resource_rep(): 1}
 
 
 class ResourceCNOT(qml.CNOT, re.ResourceOperator):
     r"""Resource class for the CNOT gate.
 
     Resources:
-        There is no further decomposition provided for this gate.
+        The CNOT gate is treated as a terminal gate and thus it cannot be decomposed
```

Comment:
Is "terminal" gate a commonly understood term?

Reply:
As in, maybe there's a better term or some kind of consensus. I'm not sure though.

Reply:
I can't really think of a better word here. There isn't really a consensus on this. Maybe fundamental?

Reply:
changed to fundamental